### PR TITLE
Track the origin of webRequest events

### PR DESCRIPTION
**Pull request by qazbnm456**
This PR is a mitigation to https://github.com/electron/electron/issues/10061, and therefore ppl can use the webContentsGetId to track the origin requester. The variable name `webContentsGetId` is taken from https://github.com/electron/electron/commit/a64978b812848e037a9ff81b1b4e2555aff8ad87.

In addition, it would be convenient if we can have a `webContents.fromGetId` api to get the corresponding `WebContents`, just like what `webContents.fromId` does. How do you think?

### CODE REVIEW

**Comment by qazbnm456** [comment]
@MarshallOfSound Done 😃

File: atom/browser/net/atom_network_delegate.cc [hunk 1]
```diff
@@ -75,9 +75,15 @@ void ToDictionary(base::DictionaryValue* details, net::URLRequest* request) {
   details->SetInteger("id", request->identifier());
   details->SetDouble("timestamp", base::Time::Now().ToDoubleT() * 1000);
   auto info = content::ResourceRequestInfo::ForRequest(request);
-  details->SetString("resourceType",
-                     info ? ResourceTypeToString(info->GetResourceType())
-                          : "other");
+  if (info) {
+    int64_t process_id = info->GetChildID();
+    int64_t routing_id = info->GetRouteID();
+    details->SetDouble("webContentsGetId", (process_id << 32) + routing_id);
```
Comment (MarshallOfSound) [inline atom/browser/net/atom_network_delegate.cc]: This should probably be just `webContentsId`

And assuming my understanding of this bit shifting is correct that ID should just work with https://electron.atom.io/docs/api/web-contents/#webcontentsfromidid 👍

Comment (qazbnm456) [reply atom/browser/net/atom_network_delegate.cc]: - `webContentsId` is actually the `ID` in a weak map, as shown [here](https://github.com/electron/electron/blob/bc39964ece14a428c4efdb23084f78737f25f681/atom/browser/api/trackable_object.h#L28).
- `webContentsGetId` is, in the other hand, a 64-bit integer composed of both the process id(high 32), and the RenderViewHost routing id(low 32).

Thus, the argument passed to `webContents.fromId(id)` should be `webContentsId`, not `webContentsGetId`. Besides, there's no api to get back the corresponding webContents from `webContentsGetId` afaik 😭 .

Comment (zcbenz) [reply atom/browser/net/atom_network_delegate.cc]: I think you can use the process ID and routing ID to get the WebContents with:

```c++
WebContents::FromRenderFrameHost(RenderFrameHost::FromID(process_id, routing_id));
```

**Comment by qazbnm456** [comment]
@zcbenz 
Thanks for your suggestions. I've used `process_id` and `render_id`, which is fetched from `info-> GetRenderFrameID()` to get the corresponding webContents through `content::WebContents::FromRenderFrameHost(content::RenderFrameHost::FromID(process_id, render_id))`, and then get the `webContentsId` by `atom::api::WebContents::GetIDFromWrappedClass(webContents)`. 😄 

By the way, I just changed `atom::api::WebContents::GetIDFromWrappedClass(webContents)` from protected to public method.

**Pull request merged**